client/js/serverstate.js

```diff
@@ -10,11 +13,11 @@ let deltaChanged = false;
 let deltaID = 0;
 let batchDepth = 0;
 
-function addWidget(widget) {
+export function addWidget(widget, instance) {
   if(widget.parent && !widgets.has(widget.parent)) {
     if(!deferredChildren[widget.parent])
       deferredChildren[widget.parent] = [];
-    deferredChildren[widget.parent].push(widget);
+    deferredChildren[widget.parent].push(widget); //TODO: save instance
```

Comment:
What does "save instance" mean?

Reply:
I added a parameter to pass in an existing instance of the widget to be used. I didn't update this portion of the function to keep track of it when deferring adding the widget. I didn't update this portion of the function because if I did, then I would need to test it. And I didn't have a need to have it support the deferment of a passed in pre-created instance.

So, I left this TODO here to indicate the implementation for fully supporting the new parameter is incomplete. Would you like to see the comment updated to clarify?

Reply:
No, I see what you mean. It's probably never necessary though, right? Why would you feed it hand-created widgets in the wrong order?

Reply:
Yes, true. we can probably just make sure unit tests create & add items in the needed order.

So, what do you want me to do about the TODO comment?

Reply:
I would remove it but doesn't really matter.

Reply:
```suggestion
    deferredChildren[widget.parent].push(widget);
```